daos/stored_recipes/test/stored_recipes_logic_test.py

```diff
@@ -0,0 +1,152 @@
+            "anim esse velit occaecat culpa. Deserunt duis anim dolor ex. Deserunt anim ullamco magna adipisicing nisi esse",
+            "dolore mollit nostrud pariatur laborum sit amet. Laborum sunt voluptate sint officia pariatur labore amet cillum",
+            "fugiat sint est non aliquip adipisicing."
+        ],
+        "title": "30 Mexican stored_recipes to pair with your margarita",
+        "url": "https://www.delicious.com.au/recipes/collections/gallery/national-margarita-day-17-mexican-recipes-to-pair-with-your-favourite-cocktail/gz9ye5qj?page=3"
+    },
+    {
+        "ingredients": [],
+        "instructions": [
+            "Cupidatat pariatur nisi incididunt mollit duis labore non. Deserunt amet consectetur aliqua aute proident",
+            "anim esse velit occaecat culpa. Deserunt duis anim dolor ex. Deserunt anim ullamco magna adipisicing nisi esse",
+            "dolore mollit nostrud pariatur laborum sit amet. Laborum sunt voluptate sint officia pariatur labore amet cillum",
+            "fugiat sint est non aliquip adipisicing."
+        ],
+        "title": "30 Mexican stored_recipes to pair with your margarita",
+        "url": "www.google.com"
+    },
+    {
+        "ingredients": [
+            "Nulla aute minim exercitation excepteur magna magna veniam ad sunt.",
+            "Eiusmod ullamco magna consectetur aliqua sint ipsum.",
+            "Cupidatat minim ea est quis eu adipisicing nulla consequat",
+            "sint sunt mollit esse reprehenderit. Anim ullamco enim id esse.",
+            "Eu proident velit Lorem magna adipisicing qui exercitation enim eiusmod elit",
+            "quis. Voluptate voluptate sit fugiat aute cupidatat ea anim voluptate sunt eu ullamco anim minim nisi."
+        ],
+        "instructions": [],
+        "title": "30 Mexican stored_recipes to pair with your margarita",
+        "url": "https://www.delicious.com.au/recipes/collections/gallery/national-margarita-day-17-mexican-recipes-to-pair-with-your-favourite-cocktail/gz9ye5qj?page=3"
+    }
+]
+
+repeated_urls = [
+    "www.mockito.com",
+    "https://www.yehudit-aviv.co.il/%D7%9E%D7%A8%D7%A7%D7%99%D7%9D/%D7%A7%D7%95%D7%A1%D7%A7%D7%95%D7%A1-%D7%9E%D7%94%D7%AA%D7%97%D7%9C%D7%94-%D7%95%D7%A2%D7%93-%D7%94%D7%A1%D7%95%D7%A3%D7%A3%D7%A3%D7%A3/",
+"https://www.delicious.com.au/recipes/collections/gallery/national-margarita-day-17-mexican-recipes-to-pair-with-your-favourite-cocktail/gz9ye5qj?page=3",
+    "www.google.com",
+"https://www.delicious.com.au/recipes/collections/gallery/national-margarita-day-17-mexican-recipes-to-pair-with-your-favourite-cocktail/gz9ye5qj?page=3"
+]
+
+
+def add_recipe(url, recipe):
+    if not stored_recipe_dao.url_in_db(url):
+        stored_recipe_dao.add_recipe_to_db(response=recipe)
+
+
+def get_recipe(url, ingred, instruct):
+    if stored_recipe_dao.url_in_db(url):
+        return stored_recipe_dao.get_recipe_from_db(url, ingred, instruct)
+
```

Comment:
too much external to product logic, why do you need it? 
in unit test you need to test product components, what you have here is kinda mix between test logic and product logic

Reply:
yes it was a terrible choice, that builder design pattern does not belong in here.